This patch provides a fallback to basic HTTP authentication when /login route
fails to avoid infinite redirect loop.
Index: nextcloud-server-20.0.4/lib/base.php
===================================================================
--- nextcloud-server-20.0.4.orig/lib/base.php	2021-03-01 18:02:39.784000000 +0100
+++ nextcloud-server-20.0.4/lib/base.php	2021-03-01 18:03:08.006000000 +0100
@@ -1033,8 +1033,12 @@
 			// Redirect to default application
 			OC_Util::redirectToDefaultPage();
 		} else {
-			// Not handled and not logged in
-			header('Location: '.\OC::$server->getURLGenerator()->linkToRouteAbsolute('core.login.showLoginForm'));
+			if ($requestPath === "/login") {
+				header('WWW-Authenticate: Basic');
+			} else {
+				// Not handled and not logged in
+				header('Location: '.\OC::$server->getURLGenerator()->linkToRouteAbsolute('core.login.showLoginForm'));
+			}
 		}
 	}
 
